test/1.2/statements/T_if_ambiguity.dml

```diff
@@ -0,0 +1,60 @@
+/*
+  © 2023 Intel Corporation
+  SPDX-License-Identifier: MPL-2.0
+*/
+dml 1.2;
+
+device test;
+
+// This tests both that GCC doesn't emit Wparantheses and the behavior of
+// the DML parser.
+// Although GCC emitting Wparantheses could be considered desirable for the
+// patterns this test exemplifies, even if braces were inserted to disambiguate
+// the generated C would be identical due to `mkCompound` collapse.
+// Thus Wparantheses must be avoided in general.
+
+import "testing.dml";
+
+data bool t = true;
+data bool f = false;
+
+method test() -> (bool ok) {
+    ok = false;
+    if ($t)
+        if ($f)
+            assert false;
+        else ok = true;
```

Comment:
would be good with some comment to show what this tests and how, like "This generates `if ...`, rather than `if ...` which gcc would dislike"

Reply:
Done